src/irs/core/s3.py

```diff
@@ -49,52 +49,31 @@ def calc_part_ranges(
 
 
 @dataclass
-class S3IDs:
+class S3Ids:
     """Container for bucket and object ID"""
 
     bucket_id: str
     object_id: str
 
 
-class StagingHandlerPort(ABC):
-    """Abstract base for object storage functionality dealing with staging."""
-
-    @abstractmethod
-    async def init_staging(self) -> None:
-        """Start staging a re-encrypted file to staging area, returns an upload id"""
-
-    @abstractmethod
-    async def stage_part(self, *, data: bytes, part_number: int) -> None:
-        """Save a file part to the staging area"""
-
-    @abstractmethod
-    async def complete_staging(self, *, parts: int) -> None:
-        """Complete the staging of a re-encrypted file"""
-
-    @abstractmethod
-    async def retrieve_parts(self, *, offset: int = 0) -> AsyncGenerator[bytes, None]:
-        """Get all parts from inbox, starting with file content at offset"""
-
-    @abstractmethod
-    async def retrieve_part(self, *, url: str, start: int, stop: int) -> bytes:
-        """Get one part from inbox by range"""
-
-
-class StagingHandler(StagingHandlerPort):
+class StagingHandler:
     """Wrapper for object storage staging functionality."""
 
+    class UploadNotInitializedError(RuntimeError):
+        """Raised when the upload ID is not set for a method that needs it."""
+
     def __init__(
         self,
         object_storage: S3ObjectStorage,
```

Comment:
We have the ObjectStorage Protocol from hexkit.

```suggestion
        object_storage: ObjectStorage,
```

Reply:
Adjusted accordingly in 35dd058